internal/resources/otelcollector/config.go

```diff
@@ -92,3 +103,19 @@ type DaemonSetConfig struct {
 	MemoryLimit       resource.Quantity
 	MemoryRequest     resource.Quantity
 }
+
+func (cfg *AgentConfig) WithAllowedPorts(ports []int32) *AgentConfig {
+	cfgCopy := *cfg
+
+	cfgCopy.allowedPorts = makePorts(ports)
+	return &cfgCopy
+
+}
+
+func makePorts(ports []int32) []intstr.IntOrString {
```

Comment:
Maybe let's do this conversion in `otelcollector.makeNetworkPolicyPorts`. There we already iterate over the ports and then we don't have to populate this weirt intstr type to this package.

Reply:
done